### DISCUSSION TITLE
Get offset in callback

### DISCUSSION THREAD

**Discussion by yazyrankulov** [gilbarbara/react-spotify-web-playback#170] [Q&A] [answered]
Hi! Thanks for such an amazing library!
I want to ask is this possible to get position (offset) of track in callback?

**Use case**

I have a list of tracks:
track1
track2
track3

Imagine that those tracks are the same. I need to add some visual representation which one of them is playing right now (border for example). If I have only id of track it's not possible to distinguish between those track. Also nextTrack and previosTracks won't help because they return array with max length equals 2 (imagine if I have 10 tracks with same id in queue). Maybe this can be done some other way?

**Comment by gilbarbara** [comment, marked as answer]
Hey @yazyrankulov 
Thanks, mate! 🎉

I don't think it's possible. The API accepts an offset for the play command, but I don't think there's a way to get the current track offset in the context.
Since this is handled by either the SDK or the API, I don't keep the offset in the state.
Look at the API [docs](https://developer.spotify.com/documentation/web-api), and if you find a way to do it, let me know.

But honestly, your use case is very unique. Having the same track repeated 10+ times is super weird. I would use the repeat mode for it.

Good luck

**Reply by yazyrankulov** [reply]
Hi, I just found a workaround by adding an event listener on prev/next buttons (`.addEventListener`).
About use case: the problem was that I can play/pause music by clicking on song outside of player and I should keed track of offset :)

**Reply by gilbarbara** [reply]
I mean the use-case for having the same track over and over.

**Reply by oliver-hekmatara** [reply]
@gilbarbara I know this is an old thread but I have related question. Would it be possible to add a callback for the next/prev clicks? I'm controlling the playback with a custom playlist component. So I can't send in a playlist URI or an array of track URIs, it must be one track URI at a time. That's why I need to manually handle next/prev, would be great if you could expose those events in the callback. 🙏

**Reply by gilbarbara** [reply]
Hey @oliver-hekmatara 
Do you have the tracks for building the custom component? You could use it to create a URI array, right?

**Reply by oliver-hekmatara** [reply]
Hi, yes I have the tracks and the solution with custom components works 👍. Is it possible to remove the default prev/next buttons with any options? Right now I'm hiding them with css.

**Reply by gilbarbara** [reply]
Not really. This library is a UI wrapper that works on top of the Spotify Web Playback SDK. 
If you want a completely different UI, it would be easier to build your own.
Hiding them with CSS is ok too.
